src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -680,6 +680,21 @@ void DataWriterImpl::InnerDataWriterListener::onWriterMatched(
     data_writer_->publisher_->publisher_listener_.on_publication_matched(data_writer_->user_datawriter_, info);
 }
 
+void DataWriterImpl::InnerDataWriterListener::on_offered_incompatible_qos(
+        RTPSWriter* /*writer*/,
+        fastdds::dds::QosPolicyId_t qos_id)
+{
+    OfferedIncompatibleQosStatus& status = data_writer_->update_offered_incompatible_qos(qos_id);
+    if (data_writer_->listener_ != nullptr)
+    {
+        data_writer_->listener_->on_offered_incompatible_qos(data_writer_->user_datawriter_, status);
+        status.total_count_change = 0u;
+    }
+
+    data_writer_->publisher_->publisher_listener_.on_offered_incompatible_qos(data_writer_->user_datawriter_, status);
```

Comment:
Should this be always called?

Reply:
No, that was part of the listener logic correction I had prepared, but ended up removing. We should have at least an 'else' clause, also on the rest of the listeners. Unfortunately that doesn't correct the logic, but should not do any harm and makes it better.

Reply:
Also added the filtering part, as the unit testing was already testing that